src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -454,17 +450,19 @@ pub fn prepare_build_cmd<'a, 'cfg>(cx: &mut Context<'a, 'cfg>, unit: &Unit<'a>)
     // necessary for that fingerprint.
     //
     // Hence, if there were some `rerun-if-changed` directives forcibly change
-    // the kind of fingerprint over to the `MtimeBased` variant where the
-    // relevant mtime is the output path of the build script.
+    // the kind of fingerprint by reinterpreting the dependencies output by the
+    // build script.
     let state = cx.build_state.clone();
     let key = (unit.pkg.package_id().clone(), unit.kind);
+    let root = unit.pkg.root().to_path_buf();
     let write_fingerprint = Work::new(move |_| {
         if let Some(output_path) = output_path {
             let outputs = state.outputs.lock().unwrap();
-            if !outputs[&key].rerun_if_changed.is_empty() {
-                let slot = MtimeSlot(Mutex::new(None));
-                fingerprint.local = LocalFingerprint::MtimeBased(slot,
-                                                                 output_path);
```

Comment:
I don't fully understand what happens here, but looks like the logic before was

```
if we need a costly fingerprint update operation:
   update_local()
       do costly operation
```

and now it is like

```
if we need a costly fingerprint update operation:
   update_local()
       if we need a costly fingerprint update operation:
           do costly operation
```


That is, looks like this `if` and `hash_busted` variable from the `update_local` function are trying to do the same thing in two different places. Perhaps we can get rid of this outer `if`?